leader-election-impl/src/main/java/com/palantir/paxos/SqlitePaxosStateLog.java

```diff
@@ -16,105 +16,89 @@
 
 package com.palantir.paxos;
 
-import java.io.IOException;
 import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
-import com.google.common.io.ByteStreams;
-import com.palantir.common.base.Throwables;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlobject.SingleValue;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.Define;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
 import com.palantir.common.persist.Persistable;
 
 public final class SqlitePaxosStateLog<V extends Persistable & Versionable> implements PaxosStateLog<V> {
-    private final Supplier<Connection> connectionSupplier;
+    private final String namespace;
```

Comment:
Is there a validation/parsing of the namespace somewhere in timelock? I wonder what restrictions sqllite places on table names.

Reply:
Yeah, we have to pass in a valid namespace which we take care of elsewhere

Reply:
What about sqlite restrictions? are all namespaces valid for sqlite tables?

Reply:
Can we have a Namespace wrapper type? so that the validation done is clear here?